### ISSUE TITLE
get_ss3_exe() function failing for some users due to SSL error unrelated to r4ss

### ISSUE THREAD

**Issue by iantaylor-NOAA** [r4ss/r4ss#927]
Thanks to @MOshima-PIFSC for pointing out that an issue that only NOAA folks (as far as I know) have been having for a few weeks with R on Windows not being able to access URLs is impacting `r4ss::get_ss3_exe()`. The function errors with the message `status was 'SSL connect error'`.

The SS3 releases are still available for manual download at https://github.com/nmfs-ost/ss3-source-code/releases/, but the convenient R function won't work for some of us until the SSL error is fixed.

If anyone has an idea for a change to r4ss that works around this, feel free to chime in. Otherwise, I'll just leave this issue up for informational purposes until the problem is resolved outside of r4ss.

**Comment by iantaylor-NOAA** [comment]
Fix from @k-doering-NOAA got this working again:

> TLDR:
> 1. Run write('CURL_SSL_BACKEND=openssl', file = "~/.Renviron", append = TRUE) in your R session.
> 2. Restart your R session
> 3. Run curl::curl_version()$ssl_version and confirm the return is OpenSSL/1.1.1m (Schannel)   (note the numbers may be different)
> 4. Try remotes::install_github() (e.g., devtools::install_github('tidyverse/dplyr') ) . It should work and continue to work in future R sessions.
> 
> Full details:
> This [curl R package vignette](https://cran.r-project.org/web/packages/curl/vignettes/windows.html) explains the 2 different backends for curl for windows users and how to switch between them. My computer defaulted to secure channel (called "Schannel"), which I discovered by calling curl::curl_version()$ssl_version
> 
> I then wrote to my .Renviron file to default to using SSL instead: write('CURL_SSL_BACKEND=openssl', file = "~/.Renviron", append = TRUE)
> 
> And then restarted my r session. I double-checked the backend being used by calling curl::curl_version()$ssl_version
> The return was "OpenSSL/1.1.1m (Schannel)", confirming OpenSSL was being used (the one not in parentheses). If you have a different version of OpenSSL, the numbers may be different for you in the return.
> 
> From there, I was able to use devtools::install_github(), and utils::download.file and jsonlite::fromJSON to download files/info from github.
> 
> I think the .Renviron file will mean the backend specified will be set as the backend to use the first time curl is used in a session, so running write('CURL_SSL_BACKEND=openssl', file = "~/.Renviron", append = TRUE) one time per computer should fix the issue.
> 
> If you wanted to switch back to the old backend, the line in the .Renviron file could be removed.

**Comment by kellijohnson-NOAA** [comment]
@iantaylor-NOAA would you be amendable to adding a check to `get_ss3_exe()` that ensures curl is installed appropriately and if it is not, then providing the directions above as a message? Another option is to try and find a way to ensure that curl is configured appropriately upon installing or loading r4ss but not everyone using r4ss is going to use this package so this latter option seems like overkill.

**Comment by e-perl-NOAA** [comment]
@iantaylor-NOAA I can start this.

**Comment by iantaylor-NOAA** [comment]
Good idea @kellijohnson-NOAA.
From what I understand, it's just NOAA people who are having the issue, so I think the ideal would be to make the warning dependent on seeing the combination of "SSL connect error" AND not having `curl::curl_version()$ssl_version include "OpenSSL". But if that's not possible, then warning everyone is fine.
